src/reducer/login.js

```diff
@@ -0,0 +1,113 @@
+import Immutable from 'immutable';
+import formurlencoded from "form-urlencoded";
+import {push} from "react-router-redux";
+import { addNotification as notify } from 'reapop';
+
+import Config from '../config';
+import Storage from '../storage';
+
+
+// create function that sends username and password
+
+const START_LOGIN = 'start_login';
+const LOGIN_SUCCESS = 'login_success';
+const LOGIN_FAILURE = 'login_failure';
+
+// action!!
+const startLogin = () => {
+    return {
+        type: START_LOGIN
+    }
+}
+
+const loginSuccess = () => {
+    return {
+        type: LOGIN_SUCCESS
+    }
+}
+
+const loginFailure = () => {
+    return {
+        type: LOGIN_FAILURE
+    }
+}
+
+const sendUsernamePassword = (email, password) => {
+    return async (dispatch) => {
+        try {
+            const fullEmail = email;
+            const fullPassword = password
+            dispatch(startLogin());
+
+            const response = await fetch(Config.API_URL + '/o/token/', {
+                method: 'POST',
+                headers: new Headers({
+                "Content-Type": "application/x-www-form-urlencoded"
+                }),
+                body: formurlencoded({
+                    username: fullEmail,
+                    password: fullPassword,
+                    grant_type: "password",
+                    client_id: "web",
+                    client_secret: Config.CLIENT_SECRET
+                })
+            });
+            
+
+            const status = await response.status;
+            const data = await response.json();
+            
+            Storage.set("token", data.access_token)  
```

Comment:
you want to set the token only if there is no error. I would do this in the `else` statement below